distribution/client/src/client.ts

```diff
@@ -151,7 +160,7 @@ class Client {
       this.runUpdates();
     });
 
-    this.reg.register().then((res, newRegistration) => {
+    this.reg.register().then(({res, newRegistration}) => {
```

Comment:
this arrow function looks fishy?

Reply:
This was actually something caught by the TypeScript conversion. From what I was reading on the interwebs, Promise resolve/reject functions are intended on only having a single value passed through.

Now it's passing a single object through.